packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -206,14 +211,30 @@ private void checkAndSetPendingOperation(String method, Result result) {
      * guarantees that this will be called and completed before any other methods are invoked.
      */
     @Override
-    public void init(Result result, List<String> requestedScopes, String hostedDomain) {
+    public void init(
+        Result result, String builderOptions, List<String> requestedScopes, String hostedDomain) {
       // We're not initialized until we receive `onConnected`.
       // If initialization fails, we'll receive `onConnectionFailed`
       checkAndSetPendingOperation(METHOD_INIT, result);
 
       try {
-        GoogleSignInOptions.Builder optionsBuilder =
-            new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN).requestEmail();
+        GoogleSignInOptions.Builder optionsBuilder;
+
+        if (builderOptions == null) builderOptions = DEFAULT_SIGN_IN;
```

Comment:
This should not be necessary, as we ensure the parameter has a non-null value on the Dart side.